.github/workflows/github-pages-deployment.yml

```diff
@@ -0,0 +1,47 @@
+name: github-pages-deployment
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+    types: [opened, synchronize, reopened]
+  release:
+    types:
+    - published
+
+defaults:
+  run:
+    shell: bash +e -x -o pipefail {0}
+
+jobs:
+  github-pages-deploy:
+    name: "Build Hugo Site and Deploy to GitHub Pages"
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_PAGES_REPO: https://github.com/cloudposse/terraform-aws-components
+      GITHUB_PAGES_PUSH_BRANCH: docs
+      GITHUB_PAGES_DIRECTORY: github_pages
+    steps:
+    - name: "Checkout source code at current commit"
+      uses: actions/checkout@v2
+    - name: "Build Hugo Static Site"
+      uses: cloudposse/docs/actions/publish-github-pages@publish-gh-pages-python
+      with:
+        CSS: "body {\n>.__slackin {\n  z-index: 2;\n}\nfont-size: 1.1rem;\n}"
+        STAGING_DIR: /tmp/staging
+        CONTENT: docs,modules
+        DEBUG: True
+        HUGO_URL: https://cloudposse.github.io/terraform-aws-components
+        HUGO_PUBLISH_DIR: docs
+        HUGO_CONFIG: hugo.config.new.yml # this variable is the file output location, not the input template location
+        HTMLTEST_CONFIG: .htmltest.new.yml # this variable is the file output location, not the input template location
+        GITHUB_PAGES_PULL_BRANCH: github-pages-docs
+        GITHUB_PAGES_HUGO_PATH: /tmp/hugo
+    - name: "Deploy Hugo Static Site to GitHub Pages"
+      uses: cloudposse/actions/github/git-push@0.30.0
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        GITHUB_ACTOR: ${{ secrets.GITHUB_ACTOR }}
+        GITHUB_REPOSITORY: env.GITHUB_PAGES_REPO
+        GIT_DIRECTORY: ./${{ env.GITHUB_PAGES_DIRECTORY }}/
+        GIT_BRANCH: ${{ env.GITHUB_PAGES_PUSH_BRANCH }}
```

Comment:
Lets move this to the composite action.

Reply:
I don't think that's possible. My understanding of `composite` actions is that they allow you to run a series of shell scripts, but I don't think they currently allow you to call other github actions.
Here's an announcement of a third-party solution for this that seems to share my understanding: https://github.com/actions/runner/issues/646#issuecomment-790007107